Be more specific about dart compile JS environment declaration option (#4341)
dart-lang/site-www: src/_includes/tools/dart-compile-js-options.md

  [Dart Package Configuration File.][]

`-D<flag>=<value>`

> : Defines an environment declaration and value pair
>   which can be accessed with 
>   [`String.fromEnvironment`][], [`int.fromEnvironment`][], 
>   [`bool.fromEnvironment`][], or [`bool.hasEnvironment`][].

Reviewer: **nit:** Consider breaking into two sentences.
```suggestion
: Defines an environment declaration and value pair.
  Access this pair with 
  [`String.fromEnvironment`][], [`int.fromEnvironment`][], 
  [`bool.fromEnvironment`][], or [`bool.hasEnvironment`][].
```
